File: 347-top-k-frequent-elements/top-k-frequent-elements.ts
function topKFrequent(nums: number[], k: number): number[] {
    const counter  = {}
    const freq = Array.from({length:nums.length + 1},()=>[])

    for (let num of nums){
        counter[num] = (counter[num] || 0) + 1
    }   
    for (let key in counter){        
        freq[counter[key]].push(Number(key))        
    }

    const result = []
    
    for (let i = nums.length; i >= 0; i--){
            for (let num of freq[i]){
                result.push(num)
                if(result.length == k){
                    return result
                }
            }
    }

    

  

};

File: 347-top-k-frequent-elements/top-k-frequent-elements.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        counter = {}
        freq = [[] for i in range(len(nums) + 1)]

        for num in nums:
            counter[num]  = counter.get(num, 0 ) + 1
        
        for num, occurences in counter.items():
            freq[occurences].append(num)
        
        result = []

        for i in range(len(freq) -1, 0, -1):
            for num in freq[i]:
                result.append(num)
                if len(result) == k:
                    return result

    
    

        
            